operator/v1alpha1/types_image_content_source_policy.go

```diff
@@ -0,0 +1,45 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImageContentSourcePolicy holds cluster-wide information about how to handle registry mirror rules.  The canonical name is `cluster`
+type ImageContentSourcePolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +required
+	Spec ImageContentSourcePolicySpec `json:"spec"`
+}
+
+type ImageContentSourcePolicySpec struct {
+	// DigestMappingRules holds the registry mirroring rules that are written to
+	// /etc/containers/registries.conf to be used by the runtime to determine which
+	// repo to fall back on, if the original repo cannot be accessed.
+	// We only support pulling by digests for mirrors, hence the name.
+	// Will also be used in cases such as offline installs/upgrades.
+	// +optional
+	DigestMappingRules DigestMappingRules `json:"digestMappingRules"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ImageContentSourcePolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ListMeta `json:"metadata"`
+	Items           []ImageContentSourcePolicy `json:"items"`
+}
+
+// DigestMappingRules holds cluster-wide information about how to handle mirros in the registries config.
+type DigestMappingRules struct {
+	// Sources are registries and/or repositories depending on the container runtime
```

Comment:
Can we/should we document that these must be repositories (a.com/b/c) to start with, before we support wildcards in the future?

Reply:
In the current implementation, there are no wildcards. There are hierarchical namespaces, i.e. you can say `example.com` to match `example.com/*/…/*`, or you can say `example.com/mirror-of-red-hat` to match `example.com/mirror-of-red-hat/*/…/*` including `example.com/mirror-of-redhat/openshift`.

There isn’t a way to express `example.com/*/mirror-of-red-hat`, and I hope we won’t need that. Do we need that?

Reply:
I mean, I don't even want to support `example.com` from this API for now.  So I would prefer to explicitly exclude that until we need it.